### ISSUE TITLE
Argument to show plot 

### ISSUE THREAD

**Issue by kwcooper** [fooof-tools/fooof#249]
Hi! An issue with the [current plot function](https://github.com/fooof-tools/fooof/blob/baed4d5970c19892c4b8dc9d2efa6de6813bd356/fooof/plts/fm.py#L28) is that there is not a call to `plt.show` to show the plot. This isn't an issue in jupyter notebooks, but if running in a script the user would have to import matplotlib and call show after the call to the fooof plotting function. 

A really quick fix would be to add an optional `show_fig=True` argument to the fooof plotting function which conditionally calls `plt.show()` at the end.

**Comment by TomDonoghue** [comment]
Hey @kwcooper - sorry for a delay on getting back to you on this!

I agree that it is a bit annoying that in some interfaces, our plotting requires an explicit call to show the plot. It would be nice if there was a clear / easy way to fix this - but we've tried this before, and it gets surprisingly / annoyingly tricky. 

For example, see this discussion on another one of our modules where we ran into trouble trying to do this:
https://github.com/bycycle-tools/bycycle/pull/78

Basically, as I remember, adding plot show to work as intended means keeping track of matplotlib's backend - but also, because we relatively often recurse across plot function calls and build up figures, it can get tricky to get this to work as intended (to show the plot only at the end, when it's built, rather than flashing it on and off). 

Because of these quirks, we are not currently planning on amending our plot procedures to do this within module, meaning the plot show call does need to be done in some interfaces. If you want to try and look into this further and propose a general solution, you are welcome to try!
